src/traces/scattergl/convert.js

```diff
@@ -128,6 +128,7 @@ proto.handlePick = function(pickResult) {
             this.color[index] :
             this.color,
         name: this.name,
+        pointIndex: index,
```

Comment:
Great. 

Now, can you add `pointIndex` to `pointcloud`, `heatmapgl` and `contourgl`? Note that for `heatmapgl` and `contourgl`, `pointIndex` will be a 2-item array.